src/rulesets/rest/2022-05-25/operation-rules.ts

```diff
@@ -141,7 +141,11 @@ const parameterCase = new OperationRule({
       "use the correct case",
       (queryParameter) => {
         const name = queryParameter.value.name;
-        if (snakeCase(name) !== name) {
+        if (
+          name.split(".").every((n) => {
+            snakeCase(n) === n;
+          })
+        ) {
```

Comment:
I'd suggest defining a function for this and unit testing it. I think snake_case on each split is good, but we probably also want to make sure we don't allow consecutive, leading or trailing dots (no `.foo`, `foo..bar`, `foo.bar.` for example).

Reply:
That's a good call, thanks!